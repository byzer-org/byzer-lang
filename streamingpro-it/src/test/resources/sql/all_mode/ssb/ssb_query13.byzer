load parquet.`/ssb/lineorder/part-00000-ee6e5c31-5f7f-4297-a666-902609272ba3-c000.snappy.parquet` as lineorder_parquet;
select * from lineorder_parquet as lineorder;

load parquet.`/ssb/dates/part-00000-b6e546cf-abb1-4d7a-a16a-eb830e3e0996-c000.snappy.parquet` as dates_parquet;
select * from dates_parquet as dates;

load parquet.`/ssb/customer/part-00000-7505a64d-b4bf-4186-874a-a9898502a51e-c000.snappy.parquet` as customer_parquet;
select * from customer_parquet as customer;

load parquet.`/ssb/part/part-00000-5f7824df-0fba-4456-8d58-68e45ebf4848-c000.snappy.parquet` as part_parquet;
select * from part_parquet as part;

load parquet.`/ssb/supplier/part-00000-4ce61878-9646-42dd-91de-7a42a01f5ec9-c000.snappy.parquet` as supplier_parquet;
select * from supplier_parquet as supplier;

select D_YEAR, S_CITY, P_BRAND, sum(LO_REVENUE) - sum(LO_SUPPLYCOST) as PROFIT
from lineorder
left join dates on LO_ORDERDATE = D_DATEKEY
left join customer on LO_CUSTKEY = C_CUSTKEY
left join supplier on LO_SUPPKEY = S_SUPPKEY
left join part on LO_PARTKEY = P_PARTKEY
where C_REGION = 'AMERICA'and S_NATION = 'UNITED STATES'
and (D_YEAR = 1997 or D_YEAR = 1998)
and P_CATEGORY = 'MFGR#14'
group by D_YEAR, S_CITY, P_BRAND
order by D_YEAR, S_CITY, P_BRAND as ssb_query13;
