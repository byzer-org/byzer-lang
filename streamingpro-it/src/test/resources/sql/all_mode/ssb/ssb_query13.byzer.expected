"D_YEAR","S_CITY","P_BRAND","PROFIT"
1997,"UNITED ST5","MFGR#1401",5635606
1997,"UNITED ST5","MFGR#1435",4297957
1997,"UNITED ST5","MFGR#1436",202583
1998,"UNITED ST5","MFGR#1406",4383018
1998,"UNITED ST5","MFGR#1416",7394361
1998,"UNITED ST7","MFGR#1424",3647687